### PR TITLE
scripts: split `setup-stack.sh`

### PR DESCRIPTION
**Pull request by georglauterbach**
# Description

Superseeds parts of #3112. Makes review easier. Split `setup-stack.sh`, did not add / remove any functionality. Basically a glorified copy-paste with minor renamings.

Fixes issue where reviewing #3112 is a chore, I guess.

I will leave #3112 open so I can gradually implement all the changes this single PR would bring. Having reviewed #3112 is good because you basically know what's coming @polarathene, so no time wasted here.

The first commit is basically just copy-paste everywhere, the rest is more renaming/minor adjustments.

This PR is the first part of a series to come about refatoring some parts of the setup with the final goal of disallowing faulty restarts.

---

This PR is **BIG**, but actually 95% copy-paste; there are **no** changes in functionality whatsoever.

## Type of change

- [x] Improvement (non-breaking change that does improve existing functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: target/scripts/helpers/index.sh [hunk 1]
```diff
@@ -10,6 +10,7 @@ function _import_scripts
   source "${PATH_TO_SCRIPTS}/accounts.sh"
   source "${PATH_TO_SCRIPTS}/aliases.sh"
   source "${PATH_TO_SCRIPTS}/change-detection.sh"
+  source "${PATH_TO_SCRIPTS}/dhparams.sh"
```
Comment (polarathene) [inline target/scripts/helpers/index.sh:13]: This files contents could probably live in the `ssl.sh` helper as it's related to that support (ciphers)

**Review by polarathene — APPROVED** [review]
Manually compared diffs before/after, LGTM :sweat_smile: :+1: 

Cheers for handling this step as a separate PR :tada:

File: target/scripts/helpers/index.sh (hunk 1 above)
Comment (georglauterbach) [reply target/scripts/helpers/index.sh:13]: I will merge this PR (to get it through), and will apply this feedback in the next, smaller PR :)

File: target/scripts/startup/setup-stack.sh [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+  _log 'trace' "${1}() registered"
+}
+
 function _setup
 {
+  while read -r FILE
+  do
+    # shellcheck source=/dev/null
+    source "${FILE}"
+  done < <(find /usr/local/bin/setup.d/ -type f)
```
Comment (casperklein) [inline target/scripts/startup/setup-stack.sh:17]: Generally bad practice - files including spaces for example will not be handled correctly. You should add `-print0` to the find command and define a delimiter `-d $'\0'` in the `read` command or just use globbing instead:

`for i in /usr/local/bin/setup.d/*; do source $i; done`

Comment (georglauterbach) [reply target/scripts/startup/setup-stack.sh:17]: See #3124.

Comment (casperklein) [reply target/scripts/startup/setup-stack.sh:17]: You are right, I overlooked that there is also a subdirectory and not just files.

File: target/scripts/helpers/index.sh (hunk 1 above)
Comment (georglauterbach) [reply target/scripts/helpers/index.sh:13]: See #3124

**Pull request merged**